Convert billboardcloud to pointcloud (and version overview draft for 0.20.0) (#11)
OpenSpace/OpenSpace-Docs: general/releases/changelogs/0.20.0/conversion.md

> # Details on Breaking Changes

Reviewer: ```suggestion
# Breaking Changes
```

Not sure about this

Author: I'm hesitant to shorten the title this way, as it sort of implies that we should list all breaking changes on this page (which I don't think we want to?)

Open for suggestions though.  My thought was that it is rather a page that summarizes what changes a user might have to go through to update to the new version (but maybe that actually does include all breaking changes? Unsure)

> 

Reviewer: ```suggestion
```
Having the text following directly after the header

The sections on this page describe some major breaking changes in version 0.20.0, together with instructions on how to adapt to the changes.

## A New Renderable for Point Clouds

> 

Reviewer: ```suggestion
```

> `RenderableBillboardsCloud` has been removed and replaced with a new renderable type called `RenderablePointCloud`. This renderable includes some bug fixes as well as updates to the specifications in the asset related to color mapping, fading and point sizing. These are made more explicit, to make it easier to use and more understandable compared to its predecessor. The new Renderable also has support for CSV files, in addition to the previous SPECK file format. For full details on the renderable, check out the new [content pages for point data](/manual/content/point-data/index).

Reviewer: ```suggestion
The `RenderableBillboardsCloud` class has been removed and replaced with a new Renderable type called `RenderablePointCloud`. This renderable includes some bug fixes as well as updates to the specifications in the asset related to color mapping, fading, and point sizing. These are made more explicit, to make it easier to use, and more understandable compared to its predecessor. The new Renderable also has support for CSV files, in addition to the previous SPECK file format. For full details on the renderable, check out the new [content pages for point data](/manual/content/point-data/index).
```

The highlighted lines show the fields whose format or name has changed. See comments in the "After" example below for details.

::::{tab-set}
:::{tab-item} Before
```{literalinclude} files/pointcloud_before.lua
:language: lua
:emphasize-lines: 4, 11-14, 16, 18-20
```
:::

:::{tab-item} After
```{literalinclude} files/pointcloud_after.lua
:language: lua
:emphasize-lines: 4, 11-25, 27-30, 32-36
```
:::
::::

:::{tip}
Note that the `Texture` field is no longer required. If a texture is excluded, the points will be drawn as circles.
:::

#### Compute `ScaleExponent` from `ScaleFactor`
The scaling of the points has been made more intuitive by changing the previous `ScaleFactor` property slightly, to make it more apparent that it affects an exponential scaling. See [details on the point data page](/manual/content/point-data/point-data.md#controlling-the-point-size), but in general, with the new `ScaleExponent` the value is used as the exponent for `10^exponent` to compute the world-scale size of the points.

The new `ScaleExponent` property can be computed from the previous `ScaleFactor` as:
:::{math}
ScaleExponent = \log_{10}(e^{ScaleFactor / 10})
:::

### Update Color Map Data Format

> 

Reviewer: ```suggestion
```

> Previously, the first and last color entry in a .cmap file was implicitly interpreted as the color to use for values that are lower or higher than the specified range. However, this was not very clear and has been made more explicit in the update by using specific keywords for these colors. Here is an example of a color map before and after this change:

Reviewer: ```suggestion
Previously, the first and last color entry in a `.cmap` file was implicitly interpreted as the color to use for values that are lower or higher than the specified range. However, this was not very clear and has been made more explicit in the update by using specific keywords for these colors. Here is an example of a color map before and after this change:
```

::::{tab-set}
:::{tab-item} Before

> ```{literalinclude} files/before.cmap

Reviewer: Maybe name this colormap_before.cmap?  Even though it is obvious it might be easier to understand when later looking at the code

:language: python
:emphasize-lines: 3, 4, 9
```
:::

:::{tab-item} After

> ```{literalinclude} files/after.cmap

Reviewer: Same here

:language: python
:emphasize-lines: 3, 4, 9
```
:::
::::

> Note that the `belowRange` and `aboveRange` colors do not have to be specified in this exact way or order. Using the keywords, the lines for these colors can be put anywhere in the file.

Reviewer: What do you mean by "not specified in this exact way"?

Author: Mmh,  I agree that this is unclear. 

I guess I mean that the capitalization does not matter

Reviewer: Ah.. I think if "in this exact way" becomes "using camelCase" or "using this capitalization" I think it will do

The color maps that we provide with our default assets have been updated and should work fine. However, any custom color maps need to be updated to comply with the new format.

:::{tip}
Note that with the 0.20.0 release, we do provide an asset with a number of common color maps, accessible by name, that might come in handy. Check out the example files for inspiration on how to use them.
:::

### Summary of Updated Properties

> 

Reviewer: ```suggestion
```

| `ColorMap` | `Coloring.ColorMapping.ColorMap` | String | |
| `UseColorMap` | `Coloring.ColorMapping.Enabled` | Boolean | |
| `ColorOption` | `Coloring.ColorMapping.Parameter` | Integer | |
| `OptionColorRange` | `Coloring.ColorMapping.ValueRange` | Vec2 | |
| `SetRangeFromData` | `Coloring.ColorMapping.SetRangeFromData` | `nil` (Trigger Property)  | |
| `UseLinearFiltering` | (Removed) |  | |
| `SizeOption` | `Sizing.SizeMapping.Parameter` | Integer | |
| `FadeInDistances` | `Fading.FadeInDistances` | Vec2 | Should now be set based on the origin of the dataset, rather than the world-space origin |
| `DisableFadeIn` | `Fading.Enabled` | Boolean | Inverted compared to the prevoius value |
| `EnablePixelSizeControl` | `Sizing.EnableMaxSizeControl` | Boolean |  |
| `BillboardMinMaxSize` | `Sizing.MaxSize` | Float | No longer a pixel value, so the value has to be updated |
| `CorrectionSizeEndDistance` | (Removed) |  | |
| `CorrectionSizeFactor` | (Removed) |  | |
| `ScaleFactor` | `Sizing.ScaleExponent` | Float | Can be computed based on the previous value, as described in the [section above](./conversion.md#compute-scaleexponent-from-scalefactor) |

> Note that the full property URI would be something like

Reviewer: ```suggestion
Note that the full property URI would be of the form
```
